salt/proxy/dummy.py

```diff
@@ -35,16 +35,14 @@ def __virtual__():
 
 
 def _save_state(details):
-    pck = open(FILENAME, 'wb')  # pylint: disable=W8470
-    pickle.dump(details, pck)
-    pck.close()
+    with open(FILENAME, 'wb') as pck:
```

Comment:
These all need to use the `salt.utils.files.fopen` function instead of Python's `open`.